compiler/modules/bir/bir.bal

```diff
@@ -92,6 +97,22 @@ public type FunctionSignature readonly & record {|
     SemType? restParamType = ();
 |};
 
+public type Region record {|
+    Label entry;
+    Label? exit = ();
+    RegionIndex? parent = ();
+    RegionKind kind;
+|};
+
+public enum RegionKind {
```

Comment:
enums consts are not scoped to the type, so these should be something like REGION_LOOP